wrapper/src/main/java/com/splunk/support/lambda/configuration/SplunkConfiguration.java

```diff
@@ -90,28 +76,27 @@ private static Level getOtelLibLogLevel() {
     return Level.WARNING;
   }
 
-  private static void setDefaultValues() {
-    setDefaultValue("otel.propagators", "b3");
-    setDefaultValue("otel.exporter", "jaeger-thrift");
-    setDefaultValue("otel.exporter.jaeger.endpoint", "http://localhost:9080/v1/trace");
-    setDefaultValue("otel.exporter.jaeger.service.name", "OtelInstrumentedLambda");
-    setDefaultValue("otel.java.disabled.resource_providers", DISABLED_RESOURCE_PROVIDERS);
-  }
+  @Override
+  public Map<String, String> getProperties() {
+    Map<String, String> config = new HashMap<>();
 
-  static void setDefaultValue(String name, String value) {
-    if (!isConfigured(name)) {
-      log.info("Setting default value. name={}, value={}", name, value);
-      System.setProperty(name, value);
-    }
-  }
+    // by default no metrics are exported
+    config.put("otel.metrics.exporter", "none");
 
-  private static boolean isConfigured(String name) {
-    return (System.getProperty(name) != null || System.getenv(toEnvVarName(name)) != null);
-  }
+    // jaeger-thrift defaults
+    config.put("otel.trace.exporter", "jaeger-thrift-splunk");
+    config.put(OTEL_EXPORTER_JAEGER_ENDPOINT, "http://localhost:9080/v1/trace");
+    config.put("otel.exporter.jaeger.service.name", "OtelInstrumentedLambda");
+
+    // B3 propagation
+    config.put("otel.propagators", "b3");
+
+    // sample ALL
+    config.put("otel.trace.sampler", "always_on");
 
-  private static final Pattern ENV_REPLACEMENT = Pattern.compile("[^a-zA-Z0-9_]");
+    // disable non-lambda resource providers
+    config.put("otel.java.disabled.resource_providers", DISABLED_RESOURCE_PROVIDERS);
```

Comment:
Name of this property has changed to `otel.java.disabled.resource-providers`

Reply:
+1